roles/confluent.variables/filter_plugins/filters.py

```diff
@@ -111,7 +112,7 @@ def split_newline_to_dict(self, string):
     def listener_properties(self, listeners_dict, default_ssl_enabled, bouncy_castle_keystore, default_ssl_mutual_auth_enabled, default_sasl_protocol,
                             kafka_broker_truststore_path, kafka_broker_truststore_storepass, kafka_broker_keystore_path, kafka_broker_keystore_storepass, kafka_broker_keystore_keypass,
                             plain_jaas_config, keytab_path, kerberos_principal, kerberos_primary,
-                            scram_user, scram_password, oauth_pem_path ):
+                            scram_user, scram_password, scram256_user, scram256_password, oauth_pem_path ):
```

Comment:
This method is getting pretty complex wrt number of arguments. And it seems like most arguments are only used conditionally. 
Not necessarily for this PR, but I would probably change this to **kwargs and simply drop a whole `kafka_properties` (or `ksql_properties` or whatever) dict as kwargs into this method.

Reply:
@stan-confluent so thinking about this, if we had a dictionary (like hostvars) that contained all of these variables, then it would be much better to just make that dictionary the argument for this function here. But I don't think that exists today, and so if we were to switch to `listener_properties(self, **kwargs)` then in all of the filter invocations i would have to update each parameter as `ssl_enabled=ssl_enabled, etc` which might even get weird w variable names.

@nerdynick do you still have intentions to create a dictionary during the variables role? Or did you come up with a better solution in https://github.com/confluentinc/cp-ansible/pull/577

Also we need to beware, I've already made a change to the filter arguments in 6.2.x, so there is starting to become discrepancies on a per branch basis. In that light you may need logic when you call the filter @stan-confluent

Reply:
Another more general point (not for this PR either) is that it is a better style to use "positive" bool checks rather than "negative" - this greatly improves readability, compare:
```python
if not omit_jaas_configs:
   # add jaas configs
if not omit_oauth_configs:
   # add oauth configs

if not omit_jaas_configs and omit_oauth_configs:
   # now here I do need a drink to understand what the condition above means :)
```
vs
```python
if inlcude_jaas_configs:  # alternative names - add_jaas_configs, jaas_enabled etc
  # add jaas configs
if include_oauth_configs:
  # add oauth configs

if include_jaas_configs and not include_oauth_configs:
  # add stuff that's only valid when oauth is enabled, but not oauth - much easier to read.
```